### DISCUSSION TITLE
Binding helper date seems broken in query

### DISCUSSION THREAD

**Discussion by deliqs** [Budibase/budibase#8806] [Help] [answered]
I'm trying to add a date via a binding for my query. I have added `{{ date now 'YYYY-MM-DD' }}` as default value. But I get this  error message `Query Error: error: invalid input syntax for type date: ""`.

When I solely select my bindings via `select {{ean}} as ean, {{ supply_date }} as date`, it results in the following:
![image](https://user-images.githubusercontent.com/88924253/203935975-82f967d3-0505-406a-8cc1-7537272bd9fe.png)
```
{
  "ean": "test",
  "date": ""
}
```

Can anybody help me out here?

**Comment by insomniacj93** [comment, marked as answer]
@deliqs Hi, what data type did you choose for "supply_date"?

**Reply by deliqs** [reply]
I don't see an option to set the data type. I set this binding in a Query in my Data Source.

**Reply by melohagan** [reply]
Also a shot in the dark, but try using double quotes instead of single quotes, i.e. `"YYYY-MM-DD"`

**Reply by deliqs** [reply]
I have already tried that to no avail.

**Reply by melohagan** [reply]
The helpers may not work for default binding values in the Data section. You may need to pass the binding through from the Design section.
A bit more on that here: https://docs.budibase.com/docs/custom-queries#using-query-bindings

So you would pass through into the *supply_date* binding the value`{{ date now "YYYY-MM-DD" }}` from the design section.

**Reply by deliqs** [reply]
@melohagan That fixes it indeed! Thanks a lot.

Do you know why the helpers are not working for default binding values? It's weird since they are shown there as an option.

**Reply by melohagan** [reply]
I think the reason is that the *Data* section represents the backend, whereas the *Design* section is the frontend. Thus using JavaScript helpers will only work from the frontend.
